python/rikai/spark/functions/vision.py

```diff
@@ -453,3 +455,36 @@ def crop(img: Image, box: Union[Box2d, List[Box2d]]):
 
     """
     return img.crop(box)
+
+
+@udf(returnType=ImageType())
+def inspect_bbox(img: Image, boxes: List[Box2d], labels: List[str]):
+    """Draw bounding boxes on the image, and the returns the inpected image
+
+    Parameters
+    ----------
+    img : Image
+        An image object to be cropped.
+    boxes : List of Box2d
+        List of bound-2d boxes.
+    labels: List of string
+        List of labels.
+
+    Returns
+    -------
+    [Image]
+        Return a list of inspected images.
+
+    Examples
+    --------
+
+    >>> spark.sql("SELECT inspect_bbox(img, boxes, labels) as bbox FROM detections")
+
+    """
+    assert len(boxes) == len(labels), "length of boxes and labeles must match"
+    pil_image = img.to_pil()
+    draw = ImageDraw.Draw(pil_image)
+    for (bbox, label) in zip(boxes, labels):
+        draw.rectangle(bbox.to_numpy().tolist(), outline="green", width=2)
+        draw.text([bbox.xmin + 5, bbox.ymin - 10], label, fill="red")
+    return Image.from_pil(pil_image)
```

Comment:
So this means that all images are loaded into memory after this point?

Reply:
Yes. To draw on images, it must be loaded into memory.

This UDF should not be used during batch processing if we care about the storage/computing cost. It is for visualization purpose.

Reply:
if we need viz specific functionality, i'm not sure a UDF is the right place to put it. Instead, I think we need the following:
1. Each annotation type should know how to draw itself on a given image
2. A function that takes a single image and a number of annotations and outputs an annotated image (using 1). I think this might be something that we have already (I vaguely remember seeing a PR from @eddyxu). 
3. A function that takes in a DataFrame and it calls 2 on the first N rows. Optionally, add some layout parameters.